### PR TITLE
Add additional tests for the repo command

### PR DESCRIPTION
**Pull request by poncovka**
* Simplify the `make-addon-pkgs.py` script.
* Test the `--includepkgs`, `--excludepkgs` and `--cost` options of the `repo` command.
* Test the `repo --name=updates` use case.

**Based on:** https://github.com/rhinstaller/kickstart-tests/pull/646 (only the last commit is relevant)

### CODE REVIEW

File: nfs-repo-and-addon.ks.in [hunk 1]
```diff
@@ -1,106 +0,0 @@
-#test name: nfs-repo-and-addon
```
Comment (VladimirSlavik) [inline nfs-repo-and-addon.ks.in:1]: Please explain (in the commit message?) why this test is deleted. I can tell that it does not fit into the set, but still...

File: proxy-auth.ks.in [hunk 2]
```diff
@@ -14,7 +14,7 @@ timezone America/New_York
 rootpw qweqwe
 shutdown
 
-# Install @core, which will also pull in testpkg-http-core from the addon repo
+# Install @core, which will also pull in a-core from the addon repo
```
Comment (VladimirSlavik) [inline proxy-auth.ks.in:17]: I think "a" is a too short name to be meaningful when reading the code for first time, "testpkg" was IMO better. But I won't fight for this :-)

File: repo-enable.ks.in [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+# Enable the repository in the installation environment.
+repo --name=test01
+
+%packages
+a-addon
+%end
+
+%pre
+
+# Create additional repositories in the installation environment.
```
Comment (VladimirSlavik) [inline repo-enable.ks.in:17]: Is that a technique that tests the functionality meaningfully? I can imagine using another disk would be the same, except for the hassle of setting it up for the test. But I am not sure what could be left out, just asking. Maybe there could be more about this in the comment(s)?

File: repo-include.ks.in [hunk 4]
```diff
@@ -0,0 +1,69 @@
+#test name: repo-include
+
+## Test for repo --cost and --includepkgs.
+
+%ksappend common/common_no_payload.ks
+%ksappend repos/default.ks
+
+# The addon repos are setup with the following packages:
```
Comment (VladimirSlavik) [inline repo-include.ks.in:8]: Thank you, this is perfect. I wonder if this explanation is the part missing in the other tests and hampering comprehension.

File: repo-include.sh [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    mkdir "${tmp_dir}/http"
+    "${PWD}/scripts/make-addon-pkgs.py" "${tmp_dir}/http"
+
+    # Start a http server to serve the test repo
+    start_httpd "${tmp_dir}/http" "${tmp_dir}"
+
+    # Substitute variables in the kickstart file.
+    sed -e "s|REPO_A_URL|${httpd_url}/a|" \
+        -e "s|REPO_B_URL|${httpd_url}/b|" \
+        "${ks}" > "${tmp_dir}/ks.cfg"
```
Comment (VladimirSlavik) [inline repo-include.sh:39]: Is this perhaps a candidate for a helper function? I have seen this at least 4 times now.

File: scripts/make-addon-pkgs.py [hunk 6]
```diff
@@ -18,28 +18,33 @@
 #
 # Red Hat Author(s): David Shea <dshea@redhat.com>
 
-# This script creates all the packages used by nfs-repo-and-addon.ks.
-# The packages are created in two directories, http and nfs. After all the rpms
-# are made just copy everything to the location set in $KSTEST_NFS_ADDON_REPO
-# and the root of the http addon repo server.
+# This script creates additional repositories for testing.
+# The packages are created in two directories, a and b.
```
Comment (VladimirSlavik) [inline scripts/make-addon-pkgs.py:22]: Are they directories, or repositories? I'd say both...

File: proxy-auth.sh [hunk 7]
```diff
@@ -24,28 +24,31 @@ TESTTYPE="method proxy knownfailure"
...
+    local httpd_url=""
+    local proxy_url=""
 
-    scriptdir=$PWD/scripts
-
-    # Create the test repo
-    PYTHONPATH=${KSTESTDIR}/lib:$PYTHONPATH ${scriptdir}/make-addon-pkgs.py $tmpdir
+    # Create the repository 'a'.
+    mkdir "${tmp_dir}/http"
+    "${PWD}/scripts/make-addon-pkgs.py" "${tmp_dir}/http"
```
Comment (VladimirSlavik) [inline proxy-auth.sh:34]: Is this a mistake, s/http/a/ ? The previous naming was quite atrocious...

**Review by VladimirSlavik — COMMENTED** [review]
Thank you, I like the idea of having a comprehensive set of tests for `repo`, it's a rather core functionality.

That said, I think the naming is not too clear. The old scheme was hard to follow, the new one is too. Perhaps it is wrong to rely on that. The repository layout could be documented comprehensively in the script for creating it, and "linked" from the individual tests with explanations which parts of that are used in the test and how.

File: repo-include.sh (hunk 5 above)
Comment (jkonecny12) [reply repo-include.sh:39]: I wonder if we want that because it's specific for a variable name so you have to have it consistent with the KS file.
But maybe that is not really a problem.

**Review by jkonecny12 — COMMENTED** [review]
Looks good to me but I would love to have this separate to multiple commits. It's hard to review this way.

File: repo-enable.sh [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+# Red Hat Author(s): David Shea <dshea@redhat.com>
```
Comment (VladimirSlavik) [inline repo-enable.sh:18]: Maybe Vendula Pocova instead?

File: repo-enable.sh [hunk 9]
```diff
@@ -0,0 +1,49 @@
+#
+# Copyright (C) 2022  Red Hat, Inc.
```
Comment (VladimirSlavik) [inline repo-enable.sh:2]: Actually, I wonder if the license is needed here, it's just another test.

File: nfs.ks.in [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+autopart
+
+keyboard us
+lang en
+timezone America/New_York
+rootpw qweqwe
+shutdown
+
+%packages
+%end
```
Comment (VladimirSlavik) [inline nfs.ks.in:17]: I wonder if some of that has a fragment, but it's really and unimportant detail, compared to the scope of this change.

**Review by VladimirSlavik — APPROVED** [review]
Thank you, the descriptions are now truly excellent.

**Review by jkonecny12 — APPROVED** [review]
Looks good to me.

File: scripts/generate-repository.py [hunk 11]
```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2015  Red Hat, Inc.
```
[the remark this answers is not in the document]
Comment (poncovka) [reply scripts/generate-repository.py]: Fixed.

File: repo-enable.sh (hunk 8 above)
Comment (poncovka) [reply repo-enable.sh]: The name is removed.

File: nfs.ks.in (hunk 10 above)
Comment (poncovka) [reply nfs.ks.in:17]: This is a leftover of the `nfs-repo-and-addon.ks.in` file, so I don't want to touch it more than is necessary.
